From 6f13f7bf0fc37cbd7a0fa075f3f5a48d9c0b192f Mon Sep 17 00:00:00 2001
From: Anson Huang <Anson.Huang@nxp.com>
Date: Tue, 20 Mar 2018 15:25:51 +0800
Subject: [PATCH 3493/5242] MLK-17843 rtc: rtc-imx-rpmsg: fix alarm enable
 function

commit  dd8115569b416c46733be597cd53cfb0242c695f from
https://source.codeaurora.org/external/imx/linux-imx.git

Add missing enable parameter for alarm enable function,
without correct parameter, the "enable" value is a random
value in memory and M4 may disable alarm unexpectedly.

Signed-off-by: Anson Huang <Anson.Huang@nxp.com>
Reviewed-by: Robin Gong <yibin.gong@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/rtc/rtc-imx-rpmsg.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/rtc/rtc-imx-rpmsg.c b/drivers/rtc/rtc-imx-rpmsg.c
index 308c1d5..82140af 100644
--- a/drivers/rtc/rtc-imx-rpmsg.c
+++ b/drivers/rtc/rtc-imx-rpmsg.c
@@ -1,5 +1,5 @@
 /*
- * Copyright 2017 NXP
+ * Copyright 2017-2018 NXP
  *
  * The code contained herein is licensed under the GNU General Public
  * License. You may obtain a copy of the GNU General Public License
@@ -199,6 +199,7 @@ static int imx_rpmsg_rtc_alarm_irq_enable(struct device *dev,
 	msg.header.minor = IMX_RMPSG_MINOR;
 	msg.header.type = RTC_RPMSG_SEND;
 	msg.header.cmd = RTC_RPMSG_ENABLE_ALARM;
+	msg.enable = enable;
 
 	ret = rtc_send_message(&msg, &rtc_rpmsg, true);
 	if (ret)
-- 
1.7.9.5

